framework/test/unit/engine/consensus/consensus.spec.ts

```diff
@@ -149,18 +151,31 @@ describe('consensus', () => {
 				logger: loggerMock,
 				db: dbMock,
 				genesisBlock: genesis,
+				legacyDB: dbMock,
 			});
 			expect(consensus['_endpoint']).toBeInstanceOf(NetworkEndpoint);
 		});
 
+		it('should instantiate legacy endpoint', async () => {
+			(chain.genesisBlockExist as jest.Mock).mockResolvedValue(true);
+			await consensus.init({
+				logger: loggerMock,
+				db: dbMock,
+				genesisBlock: genesis,
+				legacyDB: dbMock,
+			});
```

Comment:
I think we should have another issue for this improvement instead of touching other unrelated tests in this PR. Please create an issue and link it here to resolve